File: go/internal/graphql/search_by_text.go
package graphql

// SearchAnimeByTextQuery is the GraphQL query for searching anime by text
const SearchAnimeByTextQuery = `
	query ($search: String, $page: Int, $perPage: Int) {
		Page(page: $page, perPage: $perPage) {
			pageInfo {
				total
				currentPage
				lastPage
				hasNextPage
			}
			media(search: $search, type: ANIME) {
				id
				title {
					romaji
					english
					native
				}
				format
				status
				coverImage {
					large
				}
				siteUrl
			}
		}
	}`

File: src/graphql/searchByText.ts
/**
 * GraphQL query to search for anime by text
 */
export const SEARCH_ANIME_BY_TEXT = `
  query ($q: String!, $page: Int = 1, $perPage: Int = 10) {
    Page(page: $page, perPage: $perPage) {
      media(search: $q, type: ANIME, sort: [SEARCH_MATCH, POPULARITY_DESC]) {
        id
        title { romaji english native }
        format
        status
        coverImage { large }
        siteUrl
      }
      pageInfo { total currentPage lastPage hasNextPage }
    }
  }
`
